File: Solutions/Python/remove_element.py
"""
    [EASY]
    27. Remove Element

    Concepts:
    - array

    Stats:
        Runtime | 0 ms      [Beast 100%]
        Memory  | 16.58 MB  [Beats 95.82%]
"""

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        ind = 0

        while ind < len(nums):
            if nums[ind] == val:
                del nums[ind]
            else:
                ind += 1

        return len(nums)
        

File: Solutions/JavaScript/remove_element.js
/*
    [EASY]
    27. Remove Element

    Concepts:
    - array
    - two pointers

    Stats:
        Runtime | 0 ms      [Beats 100%]
        Memory  | 50.00 MB  [Beats 7.79%]
*/

/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    for (let i = 0; i < nums.length;) {
        if (nums[i] == val) {
            nums.splice(i, 1);
        } else {
            i++;
        }
    }

    return nums.length;
};